src/rust/ensogl/lib/core/src/data/color/animation.rs

```diff
@@ -70,3 +117,125 @@ impl Into<Laba> for AnimationLinearSpace<Vector4<f32>> {
         self.value.into()
     }
 }
+
+
+
+// ======================
+// === Animatable Lab ===
+// ======================
+
+impl HasAnimationSpaceRepr for Lab { type AnimationSpaceRepr = Vector3<f32>; }
+
+impl From<Lab> for AnimationLinearSpace<Vector3<f32>> {
+    fn from(value:Lab) -> AnimationLinearSpace<Vector3<f32>> {
+        let value = value.into();
+        AnimationLinearSpace { value }
+    }
+}
+
+impl Into<Lab> for AnimationLinearSpace<Vector3<f32>> {
+    fn into(self) -> Lab {
+        self.value.into()
+    }
+}
+
+
+
+// ===========
+// === Frp ===
+// ===========
+
+crate::define_endpoints! {
+    Input { }
+    Output {
+        value  (Lcha),
+    }
+}
+
+
+
+// =======================
+// === Color Animation ===
+// =======================
+
+/// The `ColorAnimation` provides color better animations for colors than the raw `Animation<_>`,
+///as it allows controlling the alpha channel separately which is important for nice fade outs.
+#[derive(Clone,CloneRef,Debug)]
+pub struct ColorAnimation {
+    initialized : Rc<Cell<bool>>,
+    frp         : Frp,
+    /// Animation of the Lch components of the color.
+    pub lch     : Animation<Lch>,
+    /// Animation of the alpha component of the color.
+    pub alpha   : Animation<f32>,
+    /// Stream of the full Lcha color.
+    pub value   : frp::Stream<Lcha>,
+}
+
+#[allow(missing_docs)]
+impl ColorAnimation {
+
+    /// Constructor.
+    pub fn new(_app:&Application) -> Self {
+        let initialized = default();
+
+        let frp         = Frp::new_network();
+        let value       = frp.output.value.clone_ref().into();
+
+        let lch         = Animation::<Lch>::new(&frp.network);
+        let alpha       = Animation::<f32>::new(&frp.network);
+
+        Self{initialized,lch,alpha,frp,value}.init()
+    }
+
+    fn init(self) -> Self {
+        let network = &self.frp.network;
+        let frp     = &self.frp;
+
+        let lch         = &self.lch;
+        let alpha       = &self.alpha;
+
+        frp::extend! { network
+
+            color_components <- all(lch.value,alpha.value);
+            color <- color_components.map(f!([]((lch,alpha)){
+                Lcha::new(lch.lightness,lch.chroma,lch.hue,*alpha)
+            }));
+
+            frp.source.value <+ color;
+        }
+        self
+    }
+
+    pub fn set_target<T:Into<Lcha>>(&self, color:T) {
+        let color = color.into();
+        if !self.initialized.replace(true) {
+            self.lch.set_value(color.opaque);
+            self.alpha.set_value(color.alpha);
+        }
+        self.lch.set_target_value(color.opaque);
+        self.alpha.set_target_value(color.alpha);
```

Comment:
Actually, that is why I'm doing it that way: otherwise we don't get the initial event that propagates the colour value.
But maybe that is the desired/specified behaviour?

Reply:
no, its good then!